CodePush.m

```diff
@@ -505,7 +506,12 @@ - (void)savePendingUpdate:(NSString *)packageHash
 RCT_EXPORT_METHOD(notifyApplicationReady:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    [CodePush removePendingUpdate];
+    // We only mark a pending update as succeeded only if it has been booted up
+    // successfully, during which `_isFirstRunAfterUpdate` is set to true.
+    if (_isFirstRunAfterUpdate) {
```

Comment:
Actually, maybe should we just on the JS side noOp notifyApplicationReady after the first time it is called? I feel like we are complicating the native notifyApplicationReady logic just to accomodate a scenarion that doesn't make sense to allow happen anyway